bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java

```diff
@@ -0,0 +1,419 @@
+                                command);
+                        if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "device is offline");
+                        }
+                    } else {
+                        if (command instanceof StringType) {
+                            final String uri = command.toFullString();
+                            volumio.replacePlay(uri, "Radio", VolumioServiceTypes.WEBRADIO);
+                        }
+                    }
+                    break;
+
+                case VolumioBindingConstants.CHANNEL_PLAY_URI:
+                    if (volumio == null) {
+                        logger.debug("Ignoring command {} = {} because device is offline.", channelUID.getId(),
+                                command);
+                        if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "device is offline");
+                        }
+                    } else {
+                        if (command instanceof StringType) {
+                            final String uri = command.toFullString();
+                            volumio.replacePlay(uri, "URI", VolumioServiceTypes.WEBRADIO);
+                        }
+                    }
+                    break;
+
+                case VolumioBindingConstants.CHANNEL_PLAY_FILE:
+                    if (volumio == null) {
+                        logger.debug("Ignoring command {} = {} because device is offline.", channelUID.getId(),
+                                command);
+                        if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "device is offline");
+                        }
+                    } else {
+                        if (command instanceof StringType) {
+                            final String uri = command.toFullString();
+                            volumio.replacePlay(uri, "", VolumioServiceTypes.MPD);
+                        }
+                    }
+                    break;
+
+                case VolumioBindingConstants.CHANNEL_PLAY_PLAYLIST:
+                    if (volumio == null) {
+                        logger.debug("Ignoring command {} = {} because device is offline.", channelUID.getId(),
+                                command);
+                        if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "device is offline");
+                        }
+                    } else {
```

Comment:
It should never get here because the same code exists at the start of the method.
To fix null analysis issues, assign `this.volumio` at the start of the method to a local variable and use that everywhere.

```suggestion
```

Reply:
solved